src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -362,8 +418,34 @@ public Task<Result> InvokeWithCancellationAsync<Result>(string targetName, IRead
         /// <exception cref="ObjectDisposedException">If this instance of <see cref="JsonRpc"/> has been disposed.</exception>
         public Task NotifyAsync(string targetName, params object[] arguments)
         {
+            // If somebody calls InvokeInternal<T>(id, "method", null), the null is not passed as an item in the array.
+            // Instead, the compiler thinks that the null is the array itself and it'll pass null directly.
+            // To account for this case, we check for null below.
+            arguments = arguments ?? new object[] { null };
```

Comment:
Rats. I'd made this comment before when we were developing the library but I guess it didn't get resolved.

Let's not propagate the pattern to `NotifyAsync`. Instead, let's apply the right pattern to NotifyAsync (with the 1 arg overload) and add the 1 arg overload to InvokeAsync as well. We can remove the offending code from InvokeAsync at a point where we feel good about breaking changes (e.g. 2.0).